include/spdlog/common.h

```diff
@@ -65,14 +65,23 @@
 #if defined(_MSC_VER) && (_MSC_VER < 1900)
     #define SPDLOG_NOEXCEPT _NOEXCEPT
     #define SPDLOG_CONSTEXPR
-    #define SPDLOG_CONSTEXPR_FUNC inline
 #else
     #define SPDLOG_NOEXCEPT noexcept
     #define SPDLOG_CONSTEXPR constexpr
-    #if __cplusplus >= 201402L
-        #define SPDLOG_CONSTEXPR_FUNC constexpr
+#endif
+
+// If building with std::format, can just use constexpr, otherwise if building with fmt
+// SPDLOG_CONSTEXPR_FUNC needs to be set the same as FMT_CONSTEXPR to avoid situations where
+// a constexpr function in spdlog could end up calling a non-constexpr function in fmt
+// depending on the compiler
+// If fmt determines it can't use constexpr, we should inline the function instead
+#ifdef SPDLOG_USE_STD_FORMAT
+    #define SPDLOG_CONSTEXPR_FUNC constexpr
+#else  // Being built with fmt
+    #if FMT_USE_CONSTEXPR
+        #define SPDLOG_CONSTEXPR_FUNC FMT_CONSTEXPR
     #else
-        #define SPDLOG_CONSTEXPR_FUNC inline
+	#define SPDLOG_CONSTEXPR_FUNC inline
```

Comment:
I needed to change this slightly to maintain the current behavior. If fmt determines it can't use constexpr then it defines `FMT_CONSTEXPR` as empty (https://github.com/gabime/spdlog/blob/cedfeeb95f3af11df7d3b1e7e0d3b86b334dc23b/include/spdlog/fmt/bundled/core.h#L92-L107) while spdlog's current behavior is to define it as `inline`

Reply:
We could probably use some preprocessor magic to detect when `FMT_CONSTEXPR` is empty and use `inline` instead in those situations, but that adds some non-trivial complexity.

Reply:
Why would it be empty?

Reply:
It gets defined as empty for some reason: https://github.com/gabime/spdlog/blob/cedfeeb95f3af11df7d3b1e7e0d3b86b334dc23b/include/spdlog/fmt/bundled/core.h#L106

Reply:
I think it won't matter for spdlog. It can be empty for spdlog as well.

Reply:
Updated based on that